### PR TITLE
create transaction functions doc

### PR DESCRIPTION
**Pull request by man-of-code**

### CODE REVIEW

**Comment by ratik21** [comment]
This PR is welcome. But i thought we were supposed to add documentation to the functions in `/lib/tx.ts` itself, no? [https://github.com/scale-it/algo-builder/blob/master/packages/algob/src/lib/tx.ts#L68](https://github.com/scale-it/algo-builder/blob/master/packages/algob/src/lib/tx.ts#L68)

**Comment by man-of-code** [comment]
@robert-zaremba Could you shed some light on this??

File: user-docs/transaction-functions.md [hunk 1]
```diff
@@ -0,0 +1,121 @@
+# Transaction Functions available in `algob`
+
+## `getSuggestedParams`
+
+This function performs the task of fetching the common transaction params from the connected network blockchain. Private chains might throw some error while executing this function, please make sure that the node progresses in that case.
```
Comment (amityadav0) [inline user-docs/transaction-functions.md:5]: how about `This function returns suggested params`.?

File: user-docs/transaction-functions.md [hunk 2]
```diff
@@ -0,0 +1,121 @@
+# Transaction Functions available in `algob`
+
+## `getSuggestedParams`
+
+This function performs the task of fetching the common transaction params from the connected network blockchain. Private chains might throw some error while executing this function, please make sure that the node progresses in that case. 
+
+This function accepts:
++ `algoc1`: an instance(object) of `Algodv2` class of `algosdk`, which is to be used to fetch the transaction params from the connected network.
```
Comment (amityadav0) [inline user-docs/transaction-functions.md:8]:
```suggestion
+ `algocl`: an instance(object) of `Algodv2` class of `algosdk`, which is to be used to fetch the transaction params from the connected network. 
```

File: user-docs/transaction-functions.md [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+
+## `getSuggestedParams`
+
+This function performs the task of fetching the common transaction params from the connected network blockchain. Private chains might throw some error while executing this function, please make sure that the node progresses in that case. 
+
+This function accepts:
++ `algoc1`: an instance(object) of `Algodv2` class of `algosdk`, which is to be used to fetch the transaction params from the connected network. 
+
+This function returns:
++ a promise on resolving which we get the desired transaction params.
```
Comment (amityadav0) [inline user-docs/transaction-functions.md:11]: what do you mean desired ?

File: user-docs/transaction-functions.md [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+
+The implementation can be found [here](https://github.com/scale-it/algo-builder/blob/master/packages/algob/src/lib/tx.ts#L8).
+
+For more information on Transaction Params, visit [this](https://developer.algorand.org/docs/reference/transactions/).
+
+## `mkTxParams`
+
+This function performs the trivial task of auto-generating the transaction params to be used by the user in creating the transaction object. This basically combines the custom params defined and provided by the user and the suggested params of the blockchain network and returns the combined params in a format which can be easily accepted by the blockchain network.
```
Comment (amityadav0) [inline user-docs/transaction-functions.md:19]:
```suggestion
This function performs the trivial task of auto-generating the transaction params to be used by the user in creating the transaction object. This combines the custom transaction params defined and provided by the user and the suggested transaction params of the blockchain network and returns the combined params in a format which can be easily accepted by the blockchain network. 
```

File: user-docs/transaction-functions.md [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+The implementation can be found [here](https://github.com/scale-it/algo-builder/blob/master/packages/algob/src/lib/tx.ts#L8).
+
+For more information on Transaction Params, visit [this](https://developer.algorand.org/docs/reference/transactions/).
+
+## `mkTxParams`
+
+This function performs the trivial task of auto-generating the transaction params to be used by the user in creating the transaction object. This basically combines the custom params defined and provided by the user and the suggested params of the blockchain network and returns the combined params in a format which can be easily accepted by the blockchain network. 
+
+This function accepts:
++ `algoc1`: an instance(object) of `Algodv2` class of `algosdk`
```
Comment (amityadav0) [inline user-docs/transaction-functions.md:22]:
```suggestion
+ `algocl`: an instance(object) of `Algodv2` class of `algosdk`
```

File: user-docs/transaction-functions.md [hunk 6]
```diff
@@ -0,0 +1,121 @@
...
++ `userParams`: a dict containing custom params defined by the user
++ `s`: an optional argument which is basically `suggestedParams`, in case this argument isn't provided the function auto-generates the suggested params using the function `getSuggestedParams`.
+
+This function returns:
++ a promise on resolving which we get the desired transaction params.
+
```
Comment (amityadav0) [inline user-docs/transaction-functions.md:28]:
```suggestion
```

File: user-docs/transaction-functions.md [hunk 7]
```diff
@@ -0,0 +1,121 @@
...
+This function performs the trivial task of auto-generating the transaction params to be used by the user in creating the transaction object. This basically combines the custom params defined and provided by the user and the suggested params of the blockchain network and returns the combined params in a format which can be easily accepted by the blockchain network. 
+
+This function accepts:
++ `algoc1`: an instance(object) of `Algodv2` class of `algosdk`
++ `userParams`: a dict containing custom params defined by the user
++ `s`: an optional argument which is basically `suggestedParams`, in case this argument isn't provided the function auto-generates the suggested params using the function `getSuggestedParams`.
+
+This function returns:
++ a promise on resolving which we get the desired transaction params.
```
Comment (amityadav0) [inline user-docs/transaction-functions.md:27]: what is desired here ?

**Review by amityadav0 — COMMENTED** [review]
i think this was supposed to be updated tx.ts file, but this one is OK too, probably instead of `user-docs` this is `dev-docs`.
or we can put this in tx.ts file.

File: user-docs/transaction-functions.md (hunk 1 above)
Comment (man-of-code) [reply user-docs/transaction-functions.md:5]: It's mentioned in the end. L11

File: user-docs/transaction-functions.md (hunk 2 above)
Comment (man-of-code) [reply user-docs/transaction-functions.md:8]: oh. will change this.

File: user-docs/transaction-functions.md (hunk 3 above)
Comment (man-of-code) [reply user-docs/transaction-functions.md:11]: I'll change this to suggested params.

File: user-docs/transaction-functions.md (hunk 7 above)
Comment (man-of-code) [reply user-docs/transaction-functions.md:27]: by desired I mean the response user is expecting. I'll change this to something more appropriate.

File: user-docs/transaction-functions.md (hunk 1 above)
Comment (amityadav0) [reply user-docs/transaction-functions.md:5]: just that fetching common transactions doesn't seem correct

Comment (robert-zaremba) [reply user-docs/transaction-functions.md:5]: I like Amit proposal. Try to make shorter sentences. Instead: "This function performs the task of fetching.." (we don't need to be that verbose) -> "Returns blockchain transaction suggested parameters".
Also, as noted in Discord - this is an API documentation, and should be in a doc strings. Here we should do more guide style description / tutorial.

File: user-docs/transaction-functions.md (hunk 2 above)
Comment (robert-zaremba) [reply user-docs/transaction-functions.md:8]: Let's simplify this:
`algocl`: an Algorand client, instance of `Algodv2`, used to communicate with a blockchain node.

File: user-docs/transaction-functions.md (hunk 3 above)
Comment (robert-zaremba) [reply user-docs/transaction-functions.md:11]: this line is not correct.
Anyway - let's remove this style of documentation and use proper API docs.

Comment (robert-zaremba) [reply user-docs/transaction-functions.md:11]: In the API  this almost doesn't require any documentation. We have types and everything is in context.

File: packages/algob/src/lib/tx.ts [hunk 8]
```diff
@@ -32,6 +40,13 @@ export async function mkTxParams (
   return s;
 }
 
+/**
+ * Returns a transaction object for ASA creation operation
```
Comment (ratik21) [inline packages/algob/src/lib/tx.ts:44]:
```suggestion
 * Returns SDK transaction object for ASA creation
```

File: packages/algob/src/lib/tx.ts [hunk 9]
```diff
@@ -32,6 +40,13 @@ export async function mkTxParams (
...
+ * @param name the name to be used for asset
```
Comment (ratik21) [inline packages/algob/src/lib/tx.ts:45]:
```suggestion
 * @param name asset name
```

File: packages/algob/src/lib/tx.ts [hunk 10]
```diff
@@ -32,6 +40,13 @@ export async function mkTxParams (
...
+ * @param asaDef the custom params to be used while creating the asset
```
Comment (ratik21) [inline packages/algob/src/lib/tx.ts:46]:
```suggestion
 * @param asaDef asset definition (passed in `/assets/asa.yaml)````

File: packages/algob/src/lib/tx.ts [hunk 11]
```diff
@@ -65,6 +80,12 @@ export function makeAssetCreateTxn (
   );
 }
 
+/**
+ * Returns a transaction object for ASA Opt-In operation
```
Comment (ratik21) [inline packages/algob/src/lib/tx.ts:84]:
```suggestion
 * Returns SDK transaction object for ASA Opt-In operation
```

**Review by ratik21 — APPROVED** [review]
Left few suggestions above.

**Pull request merged**